### PR TITLE
feat(docs): new architecture and analytics docs

### PR DESCRIPTION
**Pull request by calcsam**
Two new conceptual docs to better explain the process of creating an architecture.

### CODE REVIEW

File: docs/docs/conceptual/building-architecture.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+When architecting a professional Gatsby website, you need to choose a number of website systems: the framework (Gatsby), build & hosting provider, and other systems like headless CMS, component library, e-commerce, analytics, auth, search, forms, and so on.
```
Comment (KyleAMathews) [inline docs/docs/conceptual/building-architecture.md:1]:
```suggestion
When architecting a professional Gatsby website, you need to choose a number of systems: build & hosting provider, headless CMS, component library, e-commerce, analytics, auth, search, forms, and so on.
```

File: docs/docs/conceptual/modern-analytics.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+The most popular analytics tools used by websites are Google Analytics and Google Tag Manager, each launched in the late 2000s. But in the last few years, there's been an explosion of web analytics services that give deeper insight into visitor behavior.
+
+We've seen Gatsby teams reach for three categories of tools:
+
+Customer Data Platforms (Segment, Rudderstack) are basically an orchestration layer for different analytics systems. Instead of using Google Tag Manager to drop in a number of trackers, teams can use one tracker, and the CDP service will handle sending out analytics to those tools. CDP includes event replay, letting teams switch out marketing tools while transferring the underlying data. Using a CDP can significantly improve page performance.
```
Comment (KyleAMathews) [inline docs/docs/conceptual/modern-analytics.md:5]: link all the tools

**Review by KyleAMathews — COMMENTED** [review]
love it!

File: docs/docs/conceptual/building-architecture.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+- [Choosing Analytics Systems](/blog/conceptual/choosing-analytics-systems) (doc)
+
+- [Improving Your Site Performance](/docs/how-to/performance/improving-site-performance) (doc)
+
+- [Common Website Features](/docs/how-to/adding-common-features) (doc section)
+
+## Making Architecture Decisions Incrementally
+
+Architectures don't need to be finalized before the project starts. Sometimes, teams can build projects faster by parallelizing development and architecture.
```
Comment (KyleAMathews) [inline docs/docs/conceptual/building-architecture.md:33]:
```suggestion
Architectures don't need to be finalized before the project starts. Sometimes, teams can build projects faster by parallelizing development and architecture decisions.
```

**Review by KyleAMathews — APPROVED** [review]

**Comment by wardpeet** [comment]
@calcsam @KyleAMathews can you please follow our docs https://www.gatsbyjs.com/contributing/gatsby-style-guide/ before merging. I've reverted the PR.

**Pull request merged**